client/ayon_core/pipeline/colorspace.py

```diff
@@ -1277,7 +1277,10 @@ def set_colorspace_data_to_representation(
     # get one filename
     filename = representation["files"]
     if isinstance(filename, list):
-        filename = filename[0]
+        if filename:
+            filename = filename[0]
+        else:
+            filename = None
```

Comment:
The `filename` is needed only if `colorspace is None` so we should move it under the condition. And we should not set it to `None` as that would cause crash in `get_imageio_file_rules_colorspace_from_filepath`.